src/gui/torrentoptionsdialog.cpp

```diff
@@ -0,0 +1,368 @@
+        if (!torrent->isPrivate())
+        {
+            if (m_ui->checkDisableDHT->checkState() != Qt::PartiallyChecked)
+                torrent->setDHTDisabled(isDisableDHTChecked());
+            if (m_ui->checkDisablePEX->checkState() != Qt::PartiallyChecked)
+                torrent->setPEXDisabled(isDisablePEXChecked());
+            if (m_ui->checkDisableLSD->checkState() != Qt::PartiallyChecked)
+                torrent->setLSDDisabled(isDisableLSDChecked());
+        }
+    }
+
+    QDialog::accept();
+}
+
+void TorrentOptionsDialog::updateDownloadSpinValue(const int value)
+{
+    m_ui->spinDownloadLimit->setValue(value);
+}
+
+void TorrentOptionsDialog::updateUploadSpinValue(const int value)
+{
+    m_ui->spinUploadLimit->setValue(value);
+}
+
+void TorrentOptionsDialog::updateDownloadSliderValue(const int value)
+{
+    if (value > m_ui->sliderDownloadLimit->maximum())
+        m_ui->sliderDownloadLimit->setMaximum(value);
+    m_ui->sliderDownloadLimit->setValue(value);
+}
+
+void TorrentOptionsDialog::updateUploadSliderValue(const int value)
+{
+    if (value > m_ui->sliderUploadLimit->maximum())
+        m_ui->sliderUploadLimit->setMaximum(value);
+    m_ui->sliderUploadLimit->setValue(value);
+}
+
+int TorrentOptionsDialog::getDownloadSpeedLimit() const
+{
+    return m_ui->spinDownloadLimit->value();
+}
+
+int TorrentOptionsDialog::getUploadSpeedLimit() const
+{
+    return m_ui->spinUploadLimit->value();
+}
+
+bool TorrentOptionsDialog::useGlobalShareLimits() const
+{
+    return m_ui->radioUseGlobalShareLimits->isChecked();
+}
+
+qreal TorrentOptionsDialog::getRatio() const
+{
+    return (m_ui->radioNoLimit->isChecked() || !m_ui->checkMaxRatio->isChecked()) ? -1. : m_ui->ratioSpinBox->value();
+}
+
+int TorrentOptionsDialog::getSeedingTime() const
+{
+    return (m_ui->radioNoLimit->isChecked() || !m_ui->checkMaxTime->isChecked()) ? -1 : m_ui->timeSpinBox->value();
+}
+
+void TorrentOptionsDialog::handleRatioTypeChanged()
+{
+    m_ui->checkMaxRatio->setEnabled(m_ui->radioTorrentLimit->isChecked());
+    m_ui->checkMaxTime->setEnabled(m_ui->radioTorrentLimit->isChecked());
+
+    m_ui->ratioSpinBox->setEnabled(m_ui->radioTorrentLimit->isChecked() && m_ui->checkMaxRatio->isChecked());
+    m_ui->timeSpinBox->setEnabled(m_ui->radioTorrentLimit->isChecked() && m_ui->checkMaxTime->isChecked());
+}
+
+void TorrentOptionsDialog::enableRatioSpin()
+{
+    m_ui->ratioSpinBox->setEnabled(m_ui->checkMaxRatio->isChecked());
+}
+
+void TorrentOptionsDialog::enableTimeSpin()
+{
+    m_ui->timeSpinBox->setEnabled(m_ui->checkMaxTime->isChecked());
+}
+
+bool TorrentOptionsDialog::isDisableDHTChecked() const
+{
+    return m_ui->checkDisableDHT->isChecked();
+}
+
+bool TorrentOptionsDialog::isDisablePEXChecked() const
+{
+    return m_ui->checkDisablePEX->isChecked();
+}
+
+bool TorrentOptionsDialog::isDisableLSDChecked() const
+{
+    return m_ui->checkDisableLSD->isChecked();
+}
+
+void TorrentOptionsDialog::disableTristateCheckboxDHT()
```

Comment:
I would put many of trivial slots like this one directly in `connect()` invocation as lambdas.

Reply:
I forgot that's possible thanks.

/off topic
Do "fixup" commits help reviewing or nah?

Reply:
>/off topic
Do "fixup" commits help reviewing or nah?

I do not care. I'm looking at PR changes in its entirety, so do as you like. If I need any changes to be made in a separate commit, I will ask for it explicitly.